src/imagery/i.eodag/i.eodag.py

```diff
@@ -799,6 +829,47 @@ def products_compare(first, second):
     return search_result
 
 
+def skip_existing(output, search_result):
+    """Remove products that is already downloaded and saved in 'output' directory.
+
+    :param output: Output directory whose files will be compared with the scenes.
+    :type output: class'eodag.api.search_result.SearchResult'
+
+    :param search_results: EO products to be checked for existence in 'output' directory.
+    :type search_result: class'eodag.api.search_result.SearchResult'
+
+    :return: Sorted EO products
+    :rtype: class:'eodag.api.search_result.SearchResult'
+    """
+    filtered_result = []
+    # Check for previously downloaded scenes
+    existing_files = [os.path.join(output, f) for f in os.listdir(output)]
+    if len(existing_files) <= 1:
+        return search_result
+    # TODO: Implement a way to check if a file is completely downloaded
+    #       or if it was interrupted
+    for scene in search_result:
+        existing_file = [
+            sfile for sfile in existing_files if scene.properties["title"] in sfile
+        ]
+        if existing_file:
+            creation_time = datetime.utcfromtimestamp(
+                os.path.getctime(existing_file[0])
+            )
```

Comment:
The following should be significantly faster:

```suggestion
    output = Path(output)
    for scene in search_result:
        scene_file = output / scene.properties["title"]
        if scene_file.exists():
            creation_time = datetime.utcfromtimestamp(
                os.path.getctime(scene_file)
            )
```
In that case you do not need the _os.path.join_ and the loop in line 846. You could remove 846 all together change 847 to:

```if next(os.scandir(output), None):```, so no _existing_files_ variable needed either...
That is probably the fastest way to check if _output_ folder is empty...